### DISCUSSION TITLE
UIElement does not expose DataContext under "net8.0-windows10.0.19041" while "net8.0-desktop" does

### DISCUSSION THREAD

**Discussion by kucint** [unoplatform/uno#17281] [Q&A] [answered]
create a simple `Layout` in your project.
Try to build it.

```cs
/// <summary>
/// https://learn.microsoft.com/en-us/windows/apps/design/layout/attached-layouts
/// </summary>
internal sealed class MyLayout : NonVirtualizingLayout
{
    protected override Size MeasureOverride(NonVirtualizingLayoutContext context, Size availableSize)
    {
        foreach (UIElement element in context.Children)
        {
            object dataContext = element.DataContext; // errror
        }

        return availableSize;
    }

    protected override Size ArrangeOverride(NonVirtualizingLayoutContext context, Size finalSize)
    {
        foreach (UIElement element in context.Children)
        {
            object dataContext = element.DataContext; // errror
        }

        return finalSize;
    }
}
```

**Actual Behavior:**

- Under **net8.0-desktop**, the code compiles as expected.
- Under **net8.0-windows10.0.19041**: compilation error occurs: _error CS8917: The delegate type could not be inferred._

It seams that `UIElement.DataContext` is not exposed (is not not public).

**Expected behavior**:
 Both frameworks behave in the same way.

**Workaround**: cast the element on higher class : `object dataContext = ((FrameworkElement)element).DataContext;`
Of course, the layout must know what the class to cast to, which is not convenient.

**Comment by MartinZikmund** [comment, marked as answer]
Unfortunately, changing this is a breaking change which we cannot do between until the next major release. That being said, we are including this in the list of planned breaking changes when the time comes
